spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/repository/RedisStreamDefinitionRepository.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.admin.repository;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.cloud.data.core.StreamDefinition;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.core.BoundHashOperations;
+import org.springframework.data.redis.core.BoundZSetOperations;
+import org.springframework.data.redis.core.DefaultTypedTuple;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.core.ZSetOperations;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+
+/**
+ * A redis implementation of {@link StreamDefinitionRepository}, storing each
+ * definition as a {@literal name:dsl} mapping in a redis hash.
+ * Also stores the list of all names in a sorted set (with constant zero score), as this
+ * guarantees lexicographic order.
+ *
+ * @author Eric Bottard
+ */
+public class RedisStreamDefinitionRepository implements StreamDefinitionRepository {
+
+	private final BoundHashOperations<String, String, String> hashOperations;
+
+	private final BoundZSetOperations<String, String> zSetOperations;
```

Comment:
Well, it wouldn't be _that_ simple, as pagination is done by index, while sorted hash typically work by key. So would have to _e.g._ re-insert in a sorted hash, iterate and drop the first N, etc. Note sure it reads better.